lib/modem_info/modem_info_params.c

```diff
@@ -139,54 +139,103 @@ int modem_info_params_get(struct modem_param_info *modem)
 
 	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_NETWORK)) {
 		ret = modem_data_get(&modem->network.current_band);
-		ret += modem_data_get(&modem->network.sup_band);
-		ret += modem_data_get(&modem->network.ip_address);
-		ret += modem_data_get(&modem->network.ue_mode);
-		ret += modem_data_get(&modem->network.current_operator);
-		ret += modem_data_get(&modem->network.cellid_hex);
-		ret += modem_data_get(&modem->network.area_code);
-		ret += modem_data_get(&modem->network.lte_mode);
-		ret += modem_data_get(&modem->network.nbiot_mode);
-		ret += modem_data_get(&modem->network.gps_mode);
-		ret += modem_data_get(&modem->network.apn);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.sup_band);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.ip_address);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.ue_mode);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.current_operator);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.cellid_hex);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.area_code);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.lte_mode);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.nbiot_mode);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.gps_mode);
+		if (ret) {
+			return ret;
+		}
+		ret = modem_data_get(&modem->network.apn);
+		if (ret) {
+			return ret;
+		}
```

Comment:
It's good to have better error handling, but this also bumps the function size to over 100 lines. So I think it's time to do a little refactor. How about creating a fixed-size array (max number of params), and use existing logic for adding those that depend on kconfig. Then have a helper function that can call `modem_data_get` for each non-null entry in the array and do the error handling. The other function-calls can stay the way they are.

Reply:
I shrunk the function a slightly simpler way. Is that suitable?

Reply:
It's better, but why not use the same logic for the rest of the places that call `modem_data_get`? Does the order of the params matter? If not, you could just append to the array and do the loop at the end.

Reply:
Now, all of them are moved into one. Only exception is the one with special handling.

Reply:
Very nice. Thank you :)